Updated the documentation regarding binds options.

nuald opened the pull request:
The documentation is a little bit unclear regarding how `binds` works, I've added a note (got the same problem as [here](https://stackoverflow.com/questions/46545259/linuxkit-getty-getting-stuck) and spent too much time on this trivial issue that's obvious for moby developers, but not so obvious for the end users).

richardj-bsquare commented:
Yeah, hit that one quite hard myself... Now onto the how the hell do I make the docker binary and docker unix socket available from the docker service to the getty / sshd services (and then others as well).

justincormack commented:
I don't think the example is very helpful to what you are trying to explain. I agree it does need to be made clear though.

ijc commented:
Perhaps we could add some more fields, such as `extraBinds` which are invalid in image tags (so allowed only in yml files) and are appended to whatever the image's set is?

Might also consider having `dropBinds` to subtract them.

Or we could invent a bit more syntax for the strings in the existing bind's slice, like if they start with a `-` then the rest is considered to be a destination and we subtract anything targetting it, otherwise we append?

nuald commented:
Updating the actual tool is a little bit different task, and we have several options here (sorted by difficulty):

- Leave everything as-is, and just update the documentation (as I originally intended);
- Print a warning during image compilation like: `getty image binds added 2 mount points, but dropped 8`;
- Add the new syntax (like `extraBinds` and `dropBinds` mentioned above).

But I think it's out of scope of the ticket here. As for the example, I tried to add the most useful example (as I've seen a few comments about changing getty behaviour, and people faced the exact problem). If it's confusing, then I can just remove it. Or if you have any other examples in mind, I can replace it.

Please note that I tried to keep the documentation update as small as possible, and frankly I think that mounts/binds require a separate doc, because it's a big topic (I've seen confusion about `rootfsPropagation`, and docs are vague about it too). In that case, the example won't be needed, because we'll be able to describe how `binds` syntax actually works (`moby` reads labels and yml and generates the required JSON as a part of the compilation process).

justincormack commented:
I really don't want to merge the example as is - unless you understand the existing binds of `getty` it makes no sense to anyone.

nuald commented:
I've removed the example completely, and added a note about `org.mobyproject.config` label. I think that this would be enough to infer the binds values by the end user without any examples.
